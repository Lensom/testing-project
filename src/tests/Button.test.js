import { render, screen, fireEvent } from "@testing-library/react";
import CustomButton from "../components/Button";

describe("Testing Button component", () => {
    test("Click on button", async () => {
      const { container } = render(<CustomButton />);

      const button = screen.getByText("Plus");
      const counter = container.querySelector("p");

      expect(counter.textContent).toBe("Counter: 0");

      fireEvent.click(button)

      expect(counter.textContent).toBe("Counter: 1")
    })

    test('matches snapshot', () => {
      const { container } = render(<CustomButton />);
      expect(container).toMatchSnapshot();
    });
})

