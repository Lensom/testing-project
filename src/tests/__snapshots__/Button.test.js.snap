// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component matches snapshot 1`] = `
<div>
  <button>
    Plus
  </button>
  <p>
    Counter: 
    0
  </p>
</div>
`;
